File: scripts/extract_batch.py
import json
import sys

def extract_batch(start, count):
    try:
        with open('data/prompts.json', 'r', encoding='utf-8') as f:
            data = json.load(f)
        
        batch = data[start:start+count]
        
        # Output specifically for the LLM to read
        print(json.dumps(batch, indent=2, ensure_ascii=False))
        
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    # specific indices for this turn
    extract_batch(30, 50) 

File: scripts/extract_batch.ts
import fs from 'fs';
import path from 'path';

const filePath = path.join(process.cwd(), 'data', 'prompts.json');

function extractBatch(start: number, count: number) {
    try {
        if (!fs.existsSync(filePath)) {
            console.error(`File not found: ${filePath}`);
            return;
        }
        const content = fs.readFileSync(filePath, 'utf-8');
        const data = JSON.parse(content);
        const batch = data.slice(start, start + count);
        console.log(JSON.stringify(batch, null, 2));
    } catch (error) {
        console.error("Error extracting batch:", error);
    }
}

// Extract items 30 to 80 (50 items)
extractBatch(30, 50);

